sky/spot/recovery_strategy.py

```diff
@@ -172,8 +172,10 @@ def recover(self) -> float:
         resources = list(task.resources)[0]
         original_resources = resources
 
+        launched_cloud = handle.launched_resources.cloud
         launched_region = handle.launched_resources.region
-        new_resources = resources.copy(region=launched_region)
+        new_resources = resources.copy(cloud=launched_cloud,
```

Comment:
Q: why would specifying region only not work here?  It should work in a new task.

Maybe I'm not understanding this:
> This fixes the problem caused by https://github.com/skypilot-org/skypilot/pull/1014, which disables launching the cluster with region specified but without cloud specified.

Reply:
> Q: why would specifying region only not work here? It should work in a new task.

In #1014, we change the behavior in a new task by enforcing the user to specify the cloud when the region is specified to be consistent with the zone.

Reply:
I see. Is there a reason that has to be the case? That seems to introduce frictions:
- We can easily deduce the cloud from a region/zone
- Imagine a single-cloud user - it's clear to that user what the cloud is

This is an orthogonal issue that shouldn't block this PR.  cc @infwinston

Reply:
Those are great points! I think the reason is mainly for potential conflicts among zones in different clouds. @infwinston can say more about this.

Reply:
The reason was mainly to avoid potential confusion and future region/zone name conflicts.

For example, users may run into this typo between
`us-west-1a` and `us-west1-a` where the former is a zone in AWS, the latter one is in GCP.
If we automatically derive cloud then we might unintentionally launch a wrong one for users.
Also, it avoids potential region/zone name conflicts with other clouds.

Since specifying region/zone can be considered an advanced usage, I think it'd be better to ask more clarity from users. Does that makes sense?

Reply:
These make sense, but overall I still think deducing could be better: 
- The typo cases would be caught by confirmation 
- Conflicts can be resolved when they arise. 
- However, single-cloud users are most prevalent now and their UX problem (requiring them to type something redundant) doesn't have a workaround.